src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+            if isinstance(polyline, shapely.geometry.LineString):
+                if len(polyline.coords) < 2:
+                    raise ValueError('tried to create PolylineRegion with <2-point LineString')
+            elif isinstance(polyline, shapely.geometry.MultiLineString):
+                if len(polyline.geoms) == 0:
+                    raise ValueError('tried to create PolylineRegion from empty MultiLineString')
+                for line in polyline.geoms:
+                    assert len(line.coords) >= 2
+            else:
+                raise ValueError('tried to create PolylineRegion from non-LineString')
+            self.lineString = polyline
+            self.points = None
+        else:
+            raise ValueError('must specify points or polyline for PolylineRegion')
+
+        if not self.lineString.is_valid:
+            raise ValueError('tried to create PolylineRegion with '
+                             f'invalid LineString {self.lineString}')
+        self.segments = self.segmentsOf(self.lineString)
+        cumulativeLengths = []
+        total = 0
+        for p, q in self.segments:
+            dx, dy = p[0] - q[0], p[1] - q[1]
+            total += math.hypot(dx, dy)
+            cumulativeLengths.append(total)
+        self.cumulativeLengths = cumulativeLengths
+        if self.points is None:
+            pts = []
+            last = None
+            for p, q in self.segments:
+                if last is None or p[:2] != last[:2]:
+                    pts.append(toVector(p).coordinates)
+                pts.append(toVector(q).coordinates)
+                last = q
+            self.points = tuple(pts)
+
+    @classmethod
+    def segmentsOf(cls, lineString):
+        if isinstance(lineString, shapely.geometry.LineString):
+            segments = []
+            points = list(lineString.coords)
+            if len(points) < 2:
+                raise ValueError('LineString has fewer than 2 points')
+            last = points[0][:2]
+            for point in points[1:]:
+                segments.append((last, point))
+                last = point
+            return segments
+        elif isinstance(lineString, shapely.geometry.MultiLineString):
+            allSegments = []
+            for line in lineString.geoms:
+                allSegments.extend(cls.segmentsOf(line))
+            return allSegments
+        else:
+            raise ValueError('called segmentsOf on non-linestring')
+
+    @cached_property
+    def start(self):
+        """Get an `OrientedPoint` at the start of the polyline.
+
+        The OP's heading will be aligned with the orientation of the region, if
+        there is one (the default orientation pointing along the polyline).
+        """
+        pointA, pointB = self.segments[0]
+        if self.usingDefaultOrientation:
+            heading = headingOfSegment(pointA, pointB)
+        elif self.orientation is not None:
+            heading = self.orientation[Vector(*pointA)].yaw
```

Comment:
The orientation may not be planar, so shouldn't only take the `yaw` here. The docstring should also replace "heading" with "orientation".